lib/bolt/cli.rb

```diff
@@ -107,6 +107,15 @@ def create_option_parser(results)
           '* protocol is `ssh` by default, may be `ssh` or `winrm`',
           '* port is `22` by default for SSH, `5985` for winrm (Optional)'
         ) do |nodes|
+          # Reject simple mistakes in nodes. Failures are unclear when the
+          # arguemnt to the next (consumed) option is reported as invalid
+          #
+          # simple checking has to happen here instead of validate
+          nodes.each do |node|
```

Comment:
This doesn't appear to be a problem exclusive to `--nodes`. It seems like it applies to any option that expects a value. Hard-coding other CLI short args is a fragile way to address this.

Reply:
Agreed. I'm not sure how to address it.

It seems like an OptionParse issue not first tokenizing the entire string to find arguments. Without making a fix in the core it may only end up being somewhat messy.